designs/consolidation-price-improvement-factor.md

```diff
@@ -0,0 +1,350 @@
+# Consolidation Price Improvement Factor
+
+## Problem Statement
+
+Karpenter's consolidation logic creates excessive node churn by replacing nodes for marginal cost savings, leading to unnecessary pod cycling where workloads experience multiple pod rescheduling events per hour as nodes are repeatedly consolidated for savings as small as 5-15%.
+
+Users have reported nodes being disrupted and replaced every 5-10 minutes, causing:
+- Unnecessary pod rescheduling and application disruption
+- Reduced system reliability and availability
+- User frustration requiring workarounds (disabling spot-to-spot consolidation, extensive PDB usage, extended consolidateAfter periods)
+
+**Reference Issue**: https://github.com/aws/karpenter-provider-aws/issues/7146
+
+The core problem is that Karpenter consolidates nodes whenever a cheaper alternative exists, without requiring **meaningful cost improvement**. A node running at $0.95/hr will be immediately replaced by one at $0.90/hr (5% savings), then again by one at $0.85/hr (5.5% savings), creating a cascade of disruptions for minimal benefit.
+
+## Proposed Solution
+
+### Overview
+
+Introduce a **Price Improvement Factor** configuration option that allows users to require replacement nodes to offer **meaningful cost savings** before consolidation occurs. This factor acts as a threshold: only when `replacement_price < (current_price × factor)` will consolidation proceed.
+
+For example, a factor of **0.8** would require at least **20% cost savings**, while the default of **1.0** maintains current behavior (consolidate for any savings).
+
+### Configuration Levels
+
+The price improvement factor will be configurable at **two levels** with the following precedence:
+
+1. **NodePool-level** (highest priority) - Per-workload control
+2. **Operator-level** (fallback) - Cluster-wide default
+
+This dual-level approach provides:
+- **Simplicity for common case**: Set operator flag once, applies cluster-wide
+- **Flexibility for advanced case**: Override per-NodePool for workload-specific needs
+
+#### NodePool-Level Configuration
+
+Add field to the `Disruption` spec (follows existing pattern of `consolidateAfter`, `consolidationPolicy`):
+
+```go
+type Disruption struct {
+    ConsolidateAfter NillableDuration `json:"consolidateAfter"`
+    ConsolidationPolicy ConsolidationPolicy `json:"consolidationPolicy,omitempty"`
+    Budgets []Budget `json:"budgets,omitempty" hash:"ignore"`
+
+    // ConsolidationPriceImprovementFactor is the minimum cost savings required
+    // for consolidation to occur. Only consolidate when replacement nodes cost
+    // less than (current_price × factor).
+    //
+    // If not specified, falls back to the operator-level setting.
+    //
+    // Examples:
+    //   1.0 = Consolidate for any cost savings (legacy behavior)
+    //   0.8 = Require 20% cost savings
+    //   0.5 = Require 50% cost savings (very conservative)
+    //   0.0 = Disable price-based consolidation
```

Comment:
Sugg: reorient this around percentage, where it's an integer from [0, 100]. The kubernetes ecosystem avoids floats in APIs due to precision serialization challenges.